src/arrayMethodSort.js

```diff
@@ -4,9 +4,37 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
-  };
+  [].__proto__.sort2
+  = function(customCompareFunction = (a, b) => `${a}` > `${b}`) {
+      let compareFunction = customCompareFunction;
+
+      if (typeof compareFunction !== 'function') {
+        compareFunction = (a, b) => {
+          if (a < b) {
```

Comment:
The lines 13-21 are redundant as the default comparison function provided on line 8 already covers this logic.